crates/router/src/connector/bankofamerica/transformers.rs

```diff
@@ -40,19 +75,113 @@ impl<T>
 
 //TODO: Fill the struct with respective fields
 #[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
 pub struct BankofamericaPaymentsRequest {
-    amount: i64,
-    card: BankofamericaCard,
+    processing_information: ProcessingInformation,
+    payment_information: PaymentInformation,
+    order_information: OrderInformationWithBill,
+    client_reference_information: ClientReferenceInformation,
 }
 
 #[derive(Default, Debug, Serialize, Eq, PartialEq)]
-pub struct BankofamericaCard {
-    name: Secret<String>,
+#[serde(rename_all = "camelCase")]
+pub struct ProcessingInformation {
+    capture: bool,
+    capture_options: Option<CaptureOptions>,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct CaptureOptions {
+    capture_sequence_number: u32,
+    total_capture_count: u32,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct PaymentInformation {
+    card: Card,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Card {
     number: cards::CardNumber,
-    expiry_month: Secret<String>,
-    expiry_year: Secret<String>,
-    cvc: Secret<String>,
-    complete: bool,
+    expiration_month: Secret<String>,
+    expiration_year: Secret<String>,
+    security_code: Secret<String>,
+    #[serde(rename = "type")]
+    card_type: Option<String>,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct OrderInformationWithBill {
+    amount_details: Amount,
+    bill_to: BillTo,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Amount {
+    total_amount: String,
+    currency: String,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct BillTo {
+    first_name: Secret<String>,
+    last_name: Secret<String>,
+    address1: Secret<String>,
+    locality: String,
+    administrative_area: Secret<String>,
+    postal_code: Secret<String>,
+    country: api_enums::CountryAlpha2,
+    email: pii::Email,
+    phone_number: Secret<String>,
+}
+
+// for bankofamerica each item in Billing is mandatory
```

Comment:
Please add all mandatory fields in configs/default.rs for dynamic fields.

Reply:
Done